core/src/qrsignatures.c

```diff
@@ -0,0 +1,61 @@
+#include <unistd.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "ecdsa.h"
+#include "curves.h"
+#include "memzero.h"
+#include "nist256p1.h"
+#include "qrsignatures.h"
+#include "log.h"
+#include "config.h"
+
+
+
+// 70 is an arbitrary number more than 65(uncompressed key size.)
+#define _QR_HEX_BUF_SIZE 70
+static unsigned char _hex_buf[_QR_HEX_BUF_SIZE];
+/**
+ * Only for decoding pub key.
+ * Output is stored in _hex_buf
+ * Intentionally not writing a generic util as it's not required.
+ */
+static void hex_decode_pubkey(const char * str){
+  memzero(_hex_buf, _QR_HEX_BUF_SIZE);
+  for (size_t i = 0; i < 65 ; i++){
+    sscanf(str + 2*i, "%2hhx", _hex_buf + i);
+  }
+}
+
+bool check_signature(unsigned char * data, uint32_t data_len, unsigned char * signature){
+  if (data_len == 0){
```

Comment:
I like this. Even if `data_len` of value 0 is technically valid, it's 99% of the cases a programming error, so better bail soon. 👌